File: abstract-factory/python/src/FactoryCreator.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.VehicleFactory import VehicleFactory
from src.ShopFactory import ShopFactory

# 工厂的创建类，根据需求创建具体工厂
class FactoryCreator:
  # def __init__(self,):
    # print("FactoryCreator init")
    
  @staticmethod
  def get_factory(name):
    if name == 'vehicle':
      return VehicleFactory()
    elif name == 'shop':
      return ShopFactory()
    else:
      return None

File: abstract-factory/ts/src/FactoryCreator.ts
import { VehicleFactory } from './VehicleFactory.js'
import { ShopFactory } from './ShopFactory.js'
import { AbstractFactory } from './AbstractFactory.js'

export type FactoryType = AbstractFactory | VehicleFactory | ShopFactory | void

export class FactoryCreator {
  static getFactory(name: string): FactoryType {
    switch (name) {
      case 'vehicle':
        return new VehicleFactory()
      case 'shop':
        return new ShopFactory()
      default:
        return
    }
  }
}
